### ISSUE TITLE
Feeding live data, keeping keras in memory.

### ISSUE THREAD

**Issue by dpwacky** [keras-team/keras#11577]
Hello all, I am a first time user here, and I hope I don't offend any one with my question or if this is in the right place.

After fiddling around and searching the internet I can't find any good example to train Keras on live data. Only feed generators and stuff like that, but it is not clear how to implement that with live data, and saving the model to disk after each run, without the need for reloading all the modules, I need to have a continuous process. 

I have a very simple script and feeding it data, and it works well, so no problem with that, the only problem that I have is that I train one line at the time, so I wait for 10 or 20 seconds and then I have to train the next line that comes in, sometimes the line takes up one minute with new data.

So after each training for one line only, I save it, then the script stops, I wait for the next line, start the script again and do this over and over.

The only problem is that it is too slow, because I have to restart the python script everytime, it will start all modules again and then loads the one line from a csv file, and loads the model. 

I am searching for a way to train the model, save it (so I can export it to the predict process), and wait for the next line without the need of reloading everything that sometimes (especially on GPU) takes up 15 seconds to load and then the script is behind, the actual training is under 2 seconds, if all modules are loaded. And in the near future it will cost to much CPU power to load 50 or 100 of these scripts all together, so better to have them be a process with all modules preloaded into memory.

This is the simple script:

import numpy as np 
import pandas as pd
from keras.models import Sequential
from keras.layers import Dense, Dropout, Flatten
from keras.callbacks import ModelCheckpoint, EarlyStopping
from keras.layers.recurrent import LSTM, SimpleRNN
from keras.models import load_model
seed = 42
np.random.seed(seed)
pdata = pd.read_csv('train.csv')
features = list(pdata.columns)
features.remove('1')
features.remove('2')
features.remove('3')
features.remove('4')
features.remove('5')
features.remove('6')
features.remove('7')
features.remove('8')
features.remove('9')
features.remove('10')
featuresa = list(pdata.columns)
featuresa.remove('a')
y = pdata[features]
X = pdata[featuresa]
X_train = X
X_test = X
y_train = y
y_test = y
X_train = X_train.values
y_train = y_train.values
X_test  = X_test.values
y_test  = y_test.values
X_train = X_train.reshape((X_train.shape[0], X_train.shape[1], 1))
X_test = X_test.reshape((X_test.shape[0], X_test.shape[1], 1))
NB_EPOCHS = 5
BATCH_SIZE = 1
model = load_model('lstm.hdf5')
print('Starting training...')
history = model.fit(X_train,
                    y_train,
                    validation_data=(X_test, y_test),
                    epochs=NB_EPOCHS,
                    batch_size=BATCH_SIZE,
                    shuffle=False,
                    validation_split=1,
                    verbose=0)
yhat = model.predict(X_test, steps=1)
print yhat
model.save('lstm.hdf5')

Of course this is the training script after the first model was created with the parameters in it. But sometimes I need to discard data or run a backlog of the data since it will stop and needs to restart.

If somebody could be of assistance if this is even possible, and also sorry for if it is obvious, I am still learning python. And maybe there are some faults in the script above, but for my purpose, even that I am giving it the same validation / test data as the actual data, the data that I am getting on future predictions are pretty accurate.

But the main thing for me right now is that I want to keep it running on the GPU/CPU even when it is waiting for new data, so that the training is instant, instead of loading all the modules over and over again.  

Thank you!
